### PR TITLE
upgrade(ui): dreamy components

### PR DESCRIPTION
**Pull request by Chrissy**
updated-dreamies-in-context

This upgrades to Dreamy Components 2.0. You can read more about how the new version works [here](https://github.com/getsentry/dreamy-components), and the reasoning behind the changes [here](https://github.com/getsentry/dreamy-components/pull/19). 

This was verified in Chrome, Firefox, Safari, MSEdge and iOS.

### CODE REVIEW

File: src/sentry/static/sentry/app/views/releases/list/releaseLandingCard.jsx
Comment (billyvg) [inline src/sentry/static/sentry/app/views/releases/list/releaseLandingCard.jsx]: Should this use `space`?

File: src/sentry/static/sentry/app/views/releases/list/releaseLandingCard.jsx
Comment (billyvg) [inline src/sentry/static/sentry/app/views/releases/list/releaseLandingCard.jsx]: Could you follow-up with another PR to convert this to TS (and improve proptypes for `props.card`) since you're now a typescript GOD.

File: src/sentry/static/sentry/app/views/releases/list/releaseLanding.tsx
Comment (billyvg) [inline src/sentry/static/sentry/app/views/releases/list/releaseLanding.tsx]: You could instead, use the `withOrganization` and `withProject` HoCs

**Review by billyvg — APPROVED** [review]

**Review by dashed — COMMENTED** [review]
This looks good to me. I'll defer to Billy for the remaining feedback.

**Comment by Chrissy** [comment]
alrighty comments are addressed. gonna merge this once tests have passed. @billy I will convert that other file tomorrow.

File: src/sentry/static/sentry/app/views/releases/list/releaseLanding.tsx
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/releases/list/releaseLanding.tsx]: Can we just make this thing a styled component?

Comment (Chrissy) [reply src/sentry/static/sentry/app/views/releases/list/releaseLanding.tsx]: maybe. lemme try

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/releases/list/releaseLanding.tsx]: Something like

```tsx
const illustration = styled({data, className} => (
    <object data={data} className={className}>
        <img src={data} className={className} />
    </object>
))<Props>`
  width: 100%;
  height: 100%;
`;
```

Comment (Chrissy) [reply src/sentry/static/sentry/app/views/releases/list/releaseLanding.tsx]: tho wouldn't we want some kind of structure since there is a prop? i guess in my mind it has always felt weird to keep use a styled component when new props are involved.

Comment (Chrissy) [reply src/sentry/static/sentry/app/views/releases/list/releaseLanding.tsx]: but yeah what you wrote it definitely cleaner for sure.

Comment (Chrissy) [reply src/sentry/static/sentry/app/views/releases/list/releaseLanding.tsx]: ok right because `<props>` can check in the sc now with typescript. neat!

**Comment by Chrissy** [comment]
ok this is rtg will merge in on Monday

**Comment by Chrissy** [comment]
@EvanPurkhiser I added in your idea!

**Pull request merged**
